test/end-to-end/offer.js

```diff
@@ -1103,4 +1109,400 @@ describe("Offer endpoint tests", () => {
             });
         });
     });
+
+    describe("POST /offers/edit/:/offerId", () => {
+        let createOffer,
+            expired_test_offer,
+            grace_period_over_test_offer,
+            grace_period_valid_test_offer,
+            future_test_offer,
+            clean_test_offer;
+
+        beforeAll(async () => {
+            await Offer.deleteMany({});
+            await test_agent.del("/auth/login");
+            createOffer = async (offer) => {
+                const { _id, owner, ownerName, jobMinDuration, jobMaxDuration } = await Offer.create({
+                    ...offer,
+                    owner: test_company._id.toString(),
+                    ownerName: test_company.name,
+                    jobMinDuration: 4,
+                    jobMaxDuration: 5,
+                });
+                return {
+                    ...offer,
+                    owner: owner.toString(),
+                    ownerName,
+                    _id: _id.toString(),
+                    jobMinDuration,
+                    jobMaxDuration
+                };
+            };
+
+            expired_test_offer = await createOffer(generateTestOffer("2019-11-17", "2019-11-18"));
+
+            grace_period_over_test_offer = await createOffer(generateTestOffer(
+                (new Date(Date.now())).toISOString(),
+                (new Date(Date.now() + (2 * DAY_TO_MS))).toISOString()
+            ));
+
+            grace_period_valid_test_offer = await createOffer(generateTestOffer(
+                (new Date(Date.now())).toISOString(),
+                (new Date(Date.now() + (2 * DAY_TO_MS))).toISOString()
+            ));
+
+            future_test_offer = await createOffer(generateTestOffer(
+                (new Date(Date.now() + (2 * DAY_TO_MS))).toISOString(),
+                (new Date(Date.now() + (3 * DAY_TO_MS))).toISOString()
+            ));
+
+            clean_test_offer = await createOffer(generateTestOffer(
+                (new Date(Date.now() + (2 * DAY_TO_MS))).toISOString(),
+                (new Date(Date.now() + (3 * DAY_TO_MS))).toISOString()
+            ));
+        });
+
+        test("should fail not logged in", async () => {
+            await test_agent
+                .post("/offers/edit/notavalidid")
+                .expect(HTTPStatus.UNAUTHORIZED);
```

Comment:
Assert the error message as well (check the res.body.errors as the other tests)

Also, you should have a test with a valid id, that should fail as well